tamr_unify_client/attribute/subattribute.py

```diff
@@ -1,43 +1,38 @@
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Any, Dict, Optional
+
+from tamr_unify_client.attribute.type import AttributeType
+
+SubAttributeJson = Dict[str, Any]
+
+@dataclass(frozen=True)
 class SubAttribute:
-    """
-    An attribute which is itself a property of another attribute.
+    """An attribute which is itself a property of another attribute.
 
     See https://docs.tamr.com/reference#attribute-types
 
-    :param data: JSON data representing this attribute
-    :type data: :py:class:`dict`
+    Args:
+        name: Name of sub-attribute
+        description: Description of sub-attribute
+        type: See https://docs.tamr.com/reference#attribute-types
+        is_nullable: If this sub-attribute can be null
     """
+    name: str
+    type: AttributeType
+    is_nullable: bool
+    _json: SubAttributeJson = field(repr=False)
```

Comment:
can you explain this line and the following? i understand the previous 3

Reply:
this tells `dataclass` not to include the `_json` property when generating the `__repr__` implementation

Reply:
that's the leading underscore? what about the `foo = bar`? and `field(repr=False)`?

Reply:
no. I'm talking about `field(repr=False)`. `foo = bar` is syntax for default values in type annotations. leading underscore is just convention so users don't blindly rely on `_json`